### DISCUSSION TITLE
How stable are highlight captures after the recent standardization?

### DISCUSSION THREAD

**Discussion by camoz** [nvim-treesitter/nvim-treesitter#6513] [Q&A] [answered]
Hi,

I have a few question regarding the recent standardization of highlight captures with upstream in https://github.com/nvim-treesitter/nvim-treesitter/commit/1ae9b0e4558fe7868f8cda2db65239cfb14836d0:

1. I understand that it is useful to have them aligned with upstream, but why is it undesirable to have some additional captures like the [markdown header markers](https://github.com/nvim-treesitter/nvim-treesitter/issues/6260)? What kind of maintenance do they require? Not questioning any decisions, just trying to understand it. (I'm not familiar with the treesitter ecosystem.)
2. How "stable" are the highlight capture groups now, after standardization? E.g., very roughly, how many do you think will change per year (add/remove/rename)?

Thank you!

**Comment by clason** [comment, marked as answer]
1. Because colorschemes need to know which highlight groups to define. 
2. Stable; obviously we can't predict with absolute certainty, especially about the future. It's in the hands of Neovim now.
